### DISCUSSION TITLE
Bitbucket Cloud Response code 429 (Too Many Requests)

### DISCUSSION THREAD

**Discussion by capeflynn** [renovatebot/renovate#28234] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket Cloud

### Please tell us more about your question or problem

Is anyone else getting a ton of "Response code 429 (Too Many Requests)" errors from a self-hosted Renovate with Bitbucket Cloud? It has gotten so bad for us that the Renovate scans are useless right now.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by secustor** [comment]
Do you use caches? This reduces the number call made significantly.

**Reply by capeflynn** [reply]
I'm very new to this and assume I'm probably doing a few things not exactly right, but we are not using caches, because the Jenkins pipeline just do a `docker run` with the Renovate image. It can run on multiple build agents, but maybe there is a way to setup a shared cache that could be mounted onto the container.

Thank you very much for your help.

```sh
docker run --rm --env-file $envFile -v $WORKSPACE/renovate/config.js:/usr/src/app/config.js renovate/renovate
```

**Reply by secustor** [reply]
Take look at https://docs.renovatebot.com/examples/self-hosting/#filedirectory-usage

The first step would be to mount this directories from your container on Jenkins an then also persist it there or alternatively you can use an external Redis cache.

**Reply by capeflynn** [reply]
This is great, thank you very much. I will get started implementing this configuration.

Out of curiosity, how does caching reduce the calls to Bitbucket? I would think it would reduce calls to Maven or npm, but I don't understand how it is making so many calls to Bitbucket.

**Reply by secustor** [reply]
It also caches calls to list PR and such things. 

If you want to reduce clones on the other hand you would want to look into setting a repoCache
